syntax = "proto3";

message SmoothingOutput {
    bytes data = 1;
    float smoothingFactor = 2;
    float dest_width = 3;
    float dest_height = 4;
}

message SmoothingEmpty{
    bytes data = 1;
    float width = 2;
    float height = 3;
    float index = 4;
}

service SmoothingServices{
    rpc computeGuassianBlur(SmoothingEmpty) returns (SmoothingOutput) {}
    rpc computeBlockSmoothing(SmoothingEmpty) returns (SmoothingOutput) {}
}